src/python/pants/pantsd/service/scheduler_service.py

```diff
@@ -129,6 +141,23 @@ def _check_invalidation_globs(self, poll: bool):
         )
         self.terminate()
 
+    def _check_memory_usage(self):
+        if self._monitored_pantsd_pid is None:
+            return
+
+        try:
+            memory_usage_in_bytes = psutil.Process(self._monitored_pantsd_pid).memory_info()[0]
+            if memory_usage_in_bytes > self._max_memory_usage_in_bytes:
+                raise Exception(
```

Comment:
Could we use a more precise exception? Maybe create `TooMuchMemoryUsageException`.

I'm also confused why we raise and then immediately catch it. Are there other exceptions we expect to be raised?

Reply:
Yes, there could be others. The pid might be incorrect, psutil might fail in some other way, etc.

Given that it is immediately caught, the extra type is overkill I think.

Reply:
That's fine with me. Thanks